packages/@aws-cdk/aws-apigatewayv2/lib/common/base.ts

```diff
@@ -45,7 +50,7 @@ export abstract class StageBase extends Resource implements IStage {
    * @internal
    */
   protected _addDomainMapping(domainMapping: DomainMappingOptions) {
-    new ApiMapping(this, `${domainMapping.domainName}${domainMapping.mappingKey}`, {
+    this.apiMapping = new ApiMapping(this, `${domainMapping.domainName}${domainMapping.mappingKey}`, {
```

Comment:
Now that there's local state here, let's protect against multiple calls to `_addDomainMapping()`, by throwing an error.

Reply:
Good catch, I added a conditional with error throw. Please let me know if I need to change the error wording